core/state_processor.go

```diff
@@ -150,12 +155,25 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 					}
 
 					if acc.HasStorage() {
+					stIt, err := statedb.Snaps().StorageIterator(mpt.Hash(), accIt.Hash(), common.Hash{})
```

Comment:
the storage iterator was opened only once, for the last processed account. It needs to be opened each time an account has storage.